combat-trainer.lic

```diff
@@ -1425,15 +1426,55 @@ class AbilityProcess
     @barb_buffs = @buffs.delete('barb_buffs') || []
     echo("  @barb_buffs: #{@barb_buffs}") if $debug_mode_ct
     setup_barb_buff_flags
+
+    @paladin_use_badge = settings.paladin_use_badge
+    echo("  @kneel_khri: #{@kneel_khri}") if $debug_mode_ct
+
+    return unless @paladin_use_badge
+
+    if DRCI.wearing?('pilgrim badge')
+      @paladin_badge_get_verb = 'remove'
+      @paladin_badge_store_verb = 'wear'
+    else
+      @paladin_badge_get_verb = 'get'
+      @paladin_badge_store_verb = 'stow'
+    end
+
+    UserVars.paladin_last_badge_use ||= Time.now
+    @badge_reuse_cooldown = 1860 # 31 minutes
   end
 
   def execute(game_state)
+    check_paladin_skills(game_state)
     check_nonspell_buffs(game_state)
     false
   end
 
   private
 
+  def check_paladin_skills(game_state)
+    return unless DRStats.paladin? 
+
+    pray_badge if @paladin_use_badge
+  end
+
+  def pray_badge
```

Comment:
There looks to be quite a bit of overlap with the `pray_badge` function in `theurg.lic`. Could this be in a common file (and maybe return a bool for success/fail)?